src/index.ts

```diff
@@ -1,12 +1,22 @@
 import { onSnapshot, applySnapshot } from 'mobx-state-tree'
 
-import * as LocalStorage from './localStorageAdaptor.js'
+import AsyncLocalStorage from './asyncLocalStorage'
 
-export const persist = (name, store, options = {}) => {
+export interface IArgs {
+  (name: string, store: object, options?: IOptions): void
+}
+export interface IOptions {
+  storage?: any,
+  jsonify?: boolean,
+  readonly whitelist?: Array<string>,
+  readonly blacklist?: Array<string>
+}
+
+export const persist: IArgs = (name, store, options = {}) => {
   let {storage, jsonify, whitelist, blacklist} = options
 
   if (typeof window.localStorage !== 'undefined' && (!storage || storage === window.localStorage)) {
-    storage = LocalStorage
+    storage = AsyncLocalStorage
   }
   if (!jsonify) { jsonify = true } // default to true like mobx-persist
   const whitelistDict = arrToDict(whitelist)
```

Comment:
arggg just an error a bit below here in strict mode (line 29, can't comment on it specifically since it's not part of the diff... kind of annoying, should still be able to do that GitHub!). Not sure why it didn't pop up before, though VSCode has been weirdly lagging behind on TS updates

Reply:
typed the `_snapshot` as `any` to resolve it. though not sure that's optimal, see comment there now that I can comment there